datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -781,24 +814,30 @@ impl EquivalenceProperties {
     /// # Returns
     ///
     /// Returns a `Vec<Arc<dyn PhysicalExpr>>` containing the projected constants.
-    fn projected_constants(
-        &self,
-        mapping: &ProjectionMapping,
-    ) -> Vec<Arc<dyn PhysicalExpr>> {
+    fn projected_constants(&self, mapping: &ProjectionMapping) -> Vec<ConstExpr> {
         // First, project existing constants. For example, assume that `a + b`
         // is known to be constant. If the projection were `a as a_new`, `b as b_new`,
         // then we would project constant `a + b` as `a_new + b_new`.
         let mut projected_constants = self
             .constants
             .iter()
-            .flat_map(|expr| self.eq_group.project_expr(mapping, expr))
+            .flat_map(|const_expr| {
+                self.eq_group
```

Comment:
I wonder if you could add an API like `ConstExpr::map` that applied a function to the contained Expr.  That way this could look like

```rust
            .flat_map(|expr| expr.map(|expr| self.eq_group.project_expr(mapping, expr))))
```

Reply:
Changed in 747b69bf7.